rules/ast/ast.go

```diff
@@ -800,6 +876,53 @@ func (node *VectorArithExpr) Eval(timestamp clientmodel.Timestamp) Vector {
 	panic("Invalid vector arithmetic expression operands")
 }
 
+// resultMetric returns the metric for the given sample based on the vector
+// binary operation and the matching options.
+func (node *VectorArithExpr) resultMetric(ls, rs *Sample) clientmodel.COWMetric {
+	if len(node.matchBy) > 0 {
+		m := clientmodel.Metric{}
+		for _, ln := range node.matchBy {
+			m[ln] = ls.Metric.Metric[ln]
+		}
+		for _, ln := range node.includeLabels {
+			v, ok := ls.Metric.Metric[ln]
+			if rs != nil {
+				vr, okr := rs.Metric.Metric[ln]
+				if ok && okr {
+					panic("label available on both sides only allowed in by clause")
+				}
+				if okr {
+					v, ok = vr, okr
+				}
+			}
+			if ok {
+				m[ln] = v
+			}
+		}
+		return clientmodel.COWMetric{false, m}
+	}
+	if node.opType.shouldDropMetric() {
+		ls.Metric.Delete(clientmodel.MetricNameLabel)
+	}
+	return ls.Metric
+}
+
+// fingerprint calculates the fingerprint for the given metric based on the matching
+// options for the binary operation.
+func (node *VectorArithExpr) fingerprint(cm clientmodel.COWMetric) clientmodel.Fingerprint {
+	if len(node.matchBy) == 0 {
+		m := cm.Metric.Clone()
+		delete(m, clientmodel.MetricNameLabel)
```

Comment:
Same thing as with any other label. You get a 0 or more `some_name{} x` entries.